src/main/java/com/pingcap/tikv/TiSession.java

```diff
@@ -116,12 +122,43 @@ public synchronized ManagedChannel getChannel(String addressStr) {
     return channel;
   }
 
+  public synchronized ExecutorService getThreadPoolForIndexScan() {
+    ExecutorService res = indexScanThreadPool;
+    if (res == null) {
+      synchronized (this) {
```

Comment:
We already has a `synchroized` in the method signature, why do we need another `synchroized`?

Reply:
that's a mistake. nice catch.